### PR TITLE
[WIP] First commit with feed

### PR DESCRIPTION
**Pull request by RichieAHB**
Bit odd to PR into your branch but wanted to check with you before doing it!

Just added some basic feed stuff in, a few `CAPI` components that will potentially get swapped out for other `FrontsAPI` components but should provide a good basis for getting feeds in.

### CODE REVIEW

File: public/client-v2/src/components/CAPI/CAPISearchQuery.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// @flow
+
+import * as React from 'react';
+import capiQuery from '../../services/capiQuery';
+import Async, { type AsyncChild } from '../util/Async';
+
+type ImageAsset = {
```
Comment (Reettaphant) [inline public/client-v2/src/components/CAPI/CAPISearchQuery.js]: I'd image a lot of the types defined here, such as images and articles, are going to be used across the application (clipboard, rendering collections) so maybe they should live in a separate file?

File: public/client-v2/src/services/capiQuery.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// @flow
+
+import { qs } from '../util/qs';
+
+const API_BASE = 'https://content.guardianapis.com/';
+
+const capiQuery = (apiKey: string) => ({
```
Comment (Reettaphant) [inline public/client-v2/src/services/capiQuery.js]: Should we return appropriate errors to users in there is something wrong with capi? What do we display here if capi is not returning any search results?

File: public/client-v2/src/services/capiQuery.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  search: async (params: Object) => {
+    const response = await fetch(
+      `${API_BASE}search${qs({
+        ...params,
+        'api-key': apiKey
+      })}`
+    );
+
+    return response.json();
```
Comment (Reettaphant) [inline public/client-v2/src/services/capiQuery.js]: I was looking at the checks the old app does on capi results and it checks that an article has in an internalPageCode before showing it in the dropdown - we should probably repeat this check.

File: public/client-v2/src/services/capiQuery.js (hunk 2 above)
Comment (RichieAHB) [reply public/client-v2/src/services/capiQuery.js]: Yeah definitely, I wasn't quite sure whether we were actually going to use CAPI (for the thing you said about preview content which I hadn't thought about) so didn't do everything there. Let's work out what out plan is there and we can definitely spec this up! 👍

File: public/client-v2/src/components/Feed.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import CAPISearchQuery from './CAPI/CAPISearchQuery';
+import { type Element } from './CAPI/CAPISearchQuery';
+/* eslint-enable import/no-duplicates */
+import ScrollContainer from './ScrollContainer';
+import FeedItem from './FeedItem';
+import Loader from './Loader';
+
+// TODO: get apiKey from context (or speak directly to FrontsAPI)
+
+const getThumbnail = (_elements: Element[]) => {
```
Comment (Reettaphant) [inline public/client-v2/src/components/Feed.js]: All the logic that extract data from capi requests will probably lived in some shared file in the end as they will be used across the app but I guess we can start moving them when multiple components start making capi requests.

File: public/client-v2/src/services/capiQuery.js (hunk 3 above)
Comment (RichieAHB) [reply public/client-v2/src/services/capiQuery.js]: Is that a Front API specific thing? I don't think I've ever seen it in CAPI but yeah, we should definitely do that ... do you know what it represents?

File: public/client-v2/src/components/Feed.js (hunk 4 above)
Comment (RichieAHB) [reply public/client-v2/src/components/Feed.js]: I did do this but for some reason flow just started marking my types as `any` and I couldn't be bothered to figure it out 😞

File: public/client-v2/src/components/CAPI/CAPISearchQuery.js (hunk 1 above)
Comment (RichieAHB) [reply public/client-v2/src/components/CAPI/CAPISearchQuery.js]: See the below comment - I have some figuring out to do with flow about re-exporting types but I totally agree!

**Comment by RichieAHB** [comment]
Going to PR into master now instead.

**Pull request closed without merging**